### DISCUSSION TITLE
How do I get the Firebase Registration Token

### DISCUSSION THREAD

**Discussion by pjcjonas** [GodotNuts/GodotFirebase#320] [Q&A]
On Courier we need the firebase registration token to send notifications via firebase: https://www.courier.com/docs/guides/providers/push/firebase-fcm/?configuration-requirements#profile-requirements

This references the following section on the Firebase docs: https://firebase.google.com/docs/cloud-messaging/js/first-message#access_the_registration_token

How would I get this token in the GodotFirebase for Godot addon? I am able to send user login info to my firebase app, but what then?

**Comment by fenix-hub** [comment]
If you signup a user through GodotFirebase
https://github.com/GodotNuts/GodotFirebase/wiki/Authentication-and-User-Management#signup-with-email
You can access the registration token directly connecting to the `signup_succeeded(auth_info : Dictionary)` signal.
`auth_info` will contain an `idtoken` key

**Reply by pjcjonas** [reply]
I shall implement and let you know...  Thank you for the response! @fenix-hub

**Reply by fenix-hub** [reply]
You are welcome, let us know :)

**Reply by pjcjonas** [reply]
So when doing the sign up with this:

`Firebase.Auth.signup_with_email_and_password("pete112zxc2r@mail.com", "123456")`

I get the following response on the Auth:

```
{email:pete112zxc2r@mail.com, expiresIn:3600, idToken:eyJhbGciOiJSUzI1NiIsImtpZCI6Ijk1M{ { ............. } }6Wll4iw, kind:identitytoolkit#SignupNewUserResponse, localId:{ { ............. } }6BjiaN2, refreshToken:AOkPPWSCYIZRaxw9dB{ { ............. } }Os3jRYMisCWGsRQ}
```

And when I pass in the the response from the provider is as follows:
```
"provider": "firebase-fcm",
  "providerResponse": {
    "data": {
      "error": {
        "code": 400,
        "message": "The registration token is not a valid FCM registration token",
        "status": "INVALID_ARGUMENT",
        "details": [
          {
            "@type": "type.googleapis.com/google.firebase.fcm.v1.FcmError",
            "errorCode": "INVALID_ARGUMENT"
          }
        ]
      }
    },
    ```

**Reply by fenix-hub** [reply]
Ok so I suppose you need to deal with Google FCM APIs instead of Firebase ones to get that token.
I think it is a different set of APIs, unrelated to Firebase itself but they belong to Firebase Cloud Messaging (which is Google's Push Notification Broker). In that case I'm afraid we don't support them. Or at least I'm not aware of valid FCM tokens that Firebase API's can generate based on your Firebase users.

**Reply by fenix-hub** [reply]
I think that the registration token the docs are referring to is related to the token generated by the FCM SDK when a user authorizes to receive Push Notifications on their devices.
If that is the case, indeed you need the FCM SDK, which we don't support (yet).
As far as there are some REST APIs however, we can help you with them.
